src/main/java/com/arpnetworking/tsdcore/sinks/HttpSinkActor.java

```diff
@@ -140,13 +156,24 @@ public Receive createReceive() {
                     dispatchPending();
                 })
                 .match(PostRejected.class, rejected -> {
-                    processRejectedRequest(rejected);
-                    dispatchPending();
+                    final int attempt = rejected.getAttempt();
+                    if (attempt < _maximumRetries) {
+                        scheduleRetry(rejected.getRequest(), attempt);
+                    } else {
+                        processRejectedRequest(rejected);
```

Comment:
Not all status codes should be retryable ;)

Similar to how we have success codes, we should have retryable failure codes.